Sources/Plasma/CoreLib/hsStream.h

```diff
@@ -84,22 +84,22 @@ enum {
     virtual bool      Open(const plFileName &, const char * = "rb") = 0;
     virtual bool      Close()=0;
     virtual bool      AtEnd();
-    virtual uint32_t  Read(uint32_t byteCount, void * buffer) = 0;
-    virtual uint32_t  Write(uint32_t byteCount, const void* buffer) = 0;
-    virtual void      Skip(uint32_t deltaByteCount) = 0;
+    virtual size_t    Read(size_t byteCount, void * buffer) = 0;
+    virtual size_t    Write(size_t byteCount, const void* buffer) = 0;
+    virtual void      Skip(size_t deltaByteCount) = 0;
     virtual void      Rewind() = 0;
     virtual void      FastFwd();
-    virtual uint32_t  GetPosition() const;
-    virtual void      SetPosition(uint32_t position);
+    virtual size_t    GetPosition() const;
```

Comment:
Yeah, it's going to be a bit of a mess, but it also shouldn't impact 32-bit builds (since `size_t` is a `uint32_t` on 32-bit compilers).

My hope is that we can catch most of the cases where this is a problem for 64-bit through compiler warnings about integer downsizing.

Reply:
I'm talking specifically about cases where we won't receive warnings due to templated code. Eg something like:
```C++
template<typename T>
void WriteThingy(hsStream* S, const T& value)
{
    S->Write(sizeof(T), reinterpret_cast<const void*>(&value));
}
```

Reply:
@dpogue I just looked through all the references to these functions, and I don't see anything that would obviously break serialized files.  Have fun fixing all the type warnings this change will create 😛